packages/provider/provider/index.js

```diff
@@ -69,6 +61,34 @@ export default class Provider extends Component {
     }
   }
 
+  /**
+   * Handle the destruction of the component.
+   *
+   * @private
+   */
+  componentWillUnmount() {
+    this.mounted = false;
```

Comment:
practically, is it possible the same instance gets re-mounted?  i.e. do you want to re-set this back to true in `componentDidMount()`?

Reply:
> componentWillUnmount() is invoked immediately before a component is unmounted and destroyed. Perform any necessary cleanup in this method, such as invalidating timers, canceling network requests, or cleaning up any subscriptions that were created in componentDidMount(). **You should not call setState() here because the component will never be re-rendered.**

See React docs ^ so no.